### PR TITLE
Change IMDDimension::getName() to name.

### PR DESCRIPTION
**Pull request by quantumsteve**
Description of work.

This deprecates the python member function IMDDimension::getName(), which isn't terribly pythonic. The read-only data member IMDDimension::name has been added and should be used instead.

**To test:**

Review changes to the python interface for IMDDimension. Most other changes are just renaming that should be caught by the build servers.

There is no GitHub issue associated with this pull request.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/release/v3.9.0/framework.rst [hunk 1]
```diff
@@ -95,6 +95,8 @@ Improved
 Python
 ------
 
+- The function `IMDDimension::getName()` has been deprecated. Use the propery `IMDDimension::name` instead.
+
```
Comment (soininen) [inline docs/source/release/v3.9.0/framework.rst]: A C++ illiterate Python user might not understand what this means.

**Review by soininen — CHANGES_REQUESTED** [review]

**Comment by quantumsteve** [comment]
@soininen How would you recommend reformatting the release notes?

**Comment by soininen** [comment]
@quantumsteve 
Would
```
- The function `IMDDimension.getName()` has been deprecated. Use the attribute `IMDDimension.name` instead.
```
sound more pythonish?

**Comment by quantumsteve** [comment]
@soininen So that the style is consistent, I also updated my release notes from #18350.

**Review by soininen — APPROVED** [review]

**Comment by soininen** [comment]
Nice changes. :shipit:

**Pull request merged**